library/peripherals/stm32f10x/can.hpp

```diff
@@ -0,0 +1,763 @@
+    registers.data_a = data_a;
+    registers.data_b = data_b;
+
+    return registers;
+  }
+
+  bool ConfigureFilter([[maybe_unused]] uint32_t id,
+                       [[maybe_unused]] uint32_t mask,
+                       [[maybe_unused]] bool is_extended = false)
+  {
+    return false;
+  }
+  void EnableAcceptanceFilter()
+  {
+    // Activate filter initialization mode (Set bit 0)
+    SetFilterBankMode(FilterBankMasterControl::kInitialization);
+
+    // Deactivate filter 0 (Clear bit 0)
+    SetFilterActivationState(Filter::k0, FilterActivation::kNotActive);
+
+    // Configure filter 0 to single 32-bit scale configuration (Set bit 0)
+    SetFilterScale(Filter::k0, FilterScale::kSingle32BitScale);
+
+    // Clear filter registers to accept all messages.
+    channel_.can->sFilterRegister[0].FR1 = 0x0UL;
+    channel_.can->sFilterRegister[0].FR2 = 0x0UL;
+
+    // Set filter to mask mode
+    SetFilterType(Filter::k0, FilterType::kMask);
+
+    // Assign filter 0 to FIFO 0 (Clear bit 0)
+    SetFilterFIFOAssignment(Filter::k0, FIFOAssignment::kFIFO1);
+
+    // Activate filter 0 (Set bit 0)
+    SetFilterActivationState(Filter::k0, FilterActivation::kActive);
+
+    // Deactivate filter initialization mode (clear bit 0)
+    SetFilterBankMode(FilterBankMasterControl::kActive);
+  }
+
+  void SetFilterBankMode(FilterBankMasterControl mode)
+  {
+    channel_.can->FMR = bit::Insert(
+        channel_.can->FMR, Value(mode), FilterMaster::kInitializationMode);
+  }
+
+  void SetFilterType(bit::Mask filter, FilterType mode)
+  {
+    channel_.can->FM1R = bit::Insert(channel_.can->FM1R, Value(mode), filter);
+  }
+
+  void SetFilterScale(bit::Mask filter, FilterScale scale)
+  {
+    channel_.can->FS1R = bit::Insert(channel_.can->FS1R, Value(scale), filter);
+  }
+
+  void SetFilterFIFOAssignment(bit::Mask filter, FIFOAssignment fifo)
+  {
+    channel_.can->FFA1R = bit::Insert(channel_.can->FFA1R, Value(fifo), filter);
+  }
+
+  void SetFilterActivationState(bit::Mask filter, FilterActivation state)
+  {
+    channel_.can->FA1R = bit::Insert(channel_.can->FA1R, Value(state), filter);
+  }
+
+  bool GetMasterStatus(bit::Mask mode) const
+  {
+    return bit::Read(CAN1->MSR, mode);
+  }
+
+  const Port_t & channel_;
+};
+
+template <int port>
+inline Can & GetCan()
+{
+  if constexpr (port == 1)
+  {
+    static auto & port1_transmit_pin = GetPin<'A', 12>();
+    static auto & port1_read_pin     = GetPin<'A', 11>();
+
+    /// Predefined definition for CAN1
+    static const Can::Port_t kCan1 = {
+      .td_pin = port1_transmit_pin,
+      .rd_pin = port1_read_pin,
+      .can    = stm32f10x::CAN1,
+      .id     = sjsu::stm32f10x::SystemController::Peripherals::kCan1,
+    };
+
+    static Can can1(kCan1);
+    return can1;
+  }
+  else
+  {
+    static_assert(InvalidOption<port>,
+                  SJ2_ERROR_MESSAGE_DECORATOR(
+                      "Support CAN ports for STM32F10xx are CAN1"));
+    return GetCan<0>();
+  }
+}

```

Comment:
```suggestion
inline Can & GetCan()
{
    static auto & port_transmit_pin = GetPin<'A', 12>();
    static auto & port_read_pin     = GetPin<'A', 11>();

    /// Predefined definition for CAN
    static const Can::Port_t kCan = {
      .td_pin = port1_transmit_pin,
      .rd_pin = port1_read_pin,
      .can    = stm32f10x::CAN1,
      .id     = sjsu::stm32f10x::SystemController::Peripherals::kCan1,
    };

    static Can can(kCan);
    return can;
}
```

If there is only 1 can then you don't need to take a template parameter.